src/main/scala/org/broadinstitute/dsde/firecloud/service/TrialService.scala

```diff
@@ -177,36 +179,71 @@ final class TrialService
 
   private def enrollUser(userInfo: UserInfo): Future[PerRequestMessage] = {
     // get user's trial status, then check the current state
-    thurloeDao.getTrialStatus(userInfo.id, userInfo) flatMap { userTrialStatus =>
-      userTrialStatus match {
-        // can't determine the user's trial status; don't enroll
-        case None => Future(RequestCompleteWithErrorReport(BadRequest, "You are not eligible for a free trial."))
-        case Some(status) =>
-          status.state match {
-            // user already enrolled; don't re-enroll
-            case Some(TrialStates.Enrolled) => Future(RequestCompleteWithErrorReport(BadRequest, "You are already enrolled in a free trial."))
-            // user enabled (eligible) for trial, enroll!
-            case Some(TrialStates.Enabled) => {
-              // build the new state that we want to persist to indicate the user is enrolled
-              val now = Instant.now
-              val expirationDate = now.plus(FireCloudConfig.Trial.durationDays, ChronoUnit.DAYS)
-              val enrolledStatus = status.copy(
-                state = Some(TrialStates.Enrolled),
-                enrolledDate = now,
-                expirationDate = expirationDate
-              )
-              thurloeDao.saveTrialStatus(userInfo.id, userInfo, enrolledStatus) map { _ =>
-                // TODO: add user to free-trial billing project / create said project if necessary
-                RequestComplete(NoContent)
+    thurloeDao.getTrialStatus(userInfo.id, userInfo) flatMap {
+      // can't determine the user's trial status; don't enroll
+      case None => Future(RequestCompleteWithErrorReport(BadRequest, "You are not eligible for a free trial. (Error 10)"))
+      case Some(status) =>
+        status.state match {
+          // user already enrolled; don't re-enroll
+          case Some(TrialStates.Enrolled) => Future(RequestCompleteWithErrorReport(BadRequest, "You are already enrolled in a free trial. (Error 20)"))
+          // user enabled (eligible) for trial, enroll!
+          case Some(TrialStates.Enabled) => enrollUserInternal(userInfo, status)
+          // user in some other state; don't enroll
+          case Some(TrialStates.Disabled) => Future(RequestCompleteWithErrorReport(BadRequest, "You are not eligible for a free trial. (Error 30)"))
+          case Some(TrialStates.Terminated) => Future(RequestCompleteWithErrorReport(BadRequest, "You are not eligible for a free trial. (Error 40)"))
+          case None => Future(RequestCompleteWithErrorReport(BadRequest, "You are not eligible for a free trial. (Error 50)"))
```

Comment:
These guys are not YAGNIs, I've been using them for debugging. Will entertain removing before merge.